### DISCUSSION TITLE
How to give probabilistic reward feedback?

### DISCUSSION THREAD

**Discussion by prashantig25** [jspsych/jsPsych#1464] [Q&A]
Hi! I present my participants with two stimuli and they can respond by pressing the left or the right arrow. I want to repeat this trial 10 times but the feedback depends on the reward condition. For instance, if the reward condition is 50_50, I want the left arrow to be the correct response 5 times and the right arrow to be the correct response for the other five times. Depending on whether their keypress matches the correct response, they either win or lose points. How can I do that? Right now, I am just able to set the correct response to one of the key presses. I want it to change probabilistically. Here is the code that I have now. 
```
var practicetrial = {
        type: 'canvas-keyboard-response',
        on_start: function(practicetrial){
            practicetrial.contrast = Math.random(); practicetrial.contrast1 = jsPsych.currentTrial().contrast  + randbet(0.05,0.35);
        },
        stimulus: function(c) {
            gabor(c);
        },
        choices: [37,39],
        trial_duration: 1000,
        on_finish: function (data) {
            if (data.key_press === 37) {
                data.correct = true;
            } else {
                data.correct = false;
            }
        }
    };
    var fb = {
        type: 'html-keyboard-response', choices: jsPsych.NO_KEYS, trial_duration: 1000,
        stimulus: function () {
            var last_trial_correct = jsPsych.data.get().last(1).values()[0].correct;
            if (last_trial_correct) {
                return "<p style=\"font-size: 30px;color: #016901\">You win 1 point!</p>";
            } else {
                return "<p style=\"font-size: 30px;color: #ba0000\">You win 0 points!</p>"
            }
        }
    }
```
Suggestions regarding the same are appreciated.

**Comment by chester108** [comment]
I would assign the reward vs no reward responses ("You win 1 point", "You win 0 points") to an array and then use a jsPsych.randomization.sampleWithReplacement function to return feedback probabilistically. If you look at the documentation for randomizing with Jspsych  (link here:https://www.jspsych.org/core_library/jspsych-randomization/) there are a couple different options for how you can return stimuli a certain amount of time, with different probabilities etc
